### ISSUE TITLE
Failure loading target platform with different hosted p2 repositories

### ISSUE THREAD

**Issue by thahnen** [eclipse-tycho/tycho#2937]
## TL;DR

When consuming a target platform in Tycho 4 (it was the same issue with Tycho 3) with a unit from a repository that returns a response in **octet-stream** format, the Maven build fails as it cannot work with that content type. In comparison, Eclipse can load it directly inside the IDE without any issue.

## Expected behavior

The build succeeds like default for other units from a repository, no issue is thrown. This will be like having the same behavior inside the Eclipse IDE which can work nevertheless the response format.

## Actual behavior

I was trying this with our SonarLint for Eclipse project, I've created a specific [branch](https://github.com/SonarSource/sonarlint-eclipse/tree/protoype/tha/Tycho) containing the following information. Both "states" mentioned below are different commits (the two recent ones) if someone wants to take a look themself!

I tried consuming the PyDev plug-in which is publishing its artifacts to GitHub as a release and to SourceForge. First I started with using _GitHub_ as the repository as it seemed to be more convenient, doing so in Eclipse worked perfectly when refreshing and reloading the target platform (and when I installed the plug-in standalone and didn't consume it):

<img width="650" alt="Target-Platform_GitHub" src="https://github.com/eclipse-tycho/tycho/assets/6588381/08c873d7-3bc7-4cfa-bfcc-d1602b19c6ef">

But when running our Maven build I got the following error (the last part is German but roughly translates to "content type is not allowed in the prologue"):

<img width="1700" alt="Tycho4-Error-Content-Type" src="https://github.com/eclipse-tycho/tycho/assets/6588381/f7fe3f66-34fa-4284-83b8-51104eef8917">

Then I switched to using _SourceForge_ as the repository and again, in Eclipse everything worked as expected including refreshing and reloading the target platform (and once again installing the plug-in standalone):

<img width="625" alt="Target-Platform_SourceForge" src="https://github.com/eclipse-tycho/tycho/assets/6588381/b25f00ca-89aa-4f8f-bf26-35dcaf602d9e">

This time the Maven build succeeded and threw no error regarding some response content type. In addition to that I've checked if there is any configuration missing but I found none and therefore assume this is a bug.

We consume the target platform in our [pom.xml](https://github.com/SonarSource/sonarlint-eclipse/blob/protoype/tha/Tycho/pom.xml#L196-L236) as follows:
```xml
<plugin>
  <groupId>org.eclipse.tycho</groupId>
  <artifactId>target-platform-configuration</artifactId>
  <configuration>
    <target>
      <!-- this one consumes another target platform containing changes I mentioned above -->
      <file>../target-platforms/build.target</file>
    </target>
    <environments>
      <!-- here are some environments, nothing fancy -->
    </environments>
  </configuration>
</plugin>
```

## Edit 18.10.2023

Ping me when trying to fix the issue, I'd like to contribute if possible (and only for adding an integration test) 😄

**Comment by thahnen** [comment]
[stacktrace.txt](https://github.com/eclipse-tycho/tycho/files/13045680/stacktrace.txt)

**Comment by laeubi** [comment]
@thahnen interesting issue, can you provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue so we have a reproducer before working on a fix?

**Comment by laeubi** [comment]
Might be the same issue as described here:
- https://github.com/eclipse-tycho/tycho/issues/2938

**Comment by thahnen** [comment]
I'm not sure if it is the same issue, looks similar indeed. I'll provide a IT and then we can check if it is already fixed by the PR changes done for the issue you mentioned.
